channel.go

```diff
@@ -183,11 +183,56 @@ func (ch *Channel) Delete() error {
 	return ch.client.makeRequest(http.MethodDelete, p, nil, nil, nil)
 }
 
+type truncateOptions struct {
+	HardDelete  bool       `json:"hard_delete,omitempty"`
+	SkipPush    bool       `json:"skip_push,omitempty"`
+	TruncatedAt *time.Time `json:"truncated_at,omitempty"`
+	Message     *Message   `json:"message,omitempty"`
+}
+
+type TruncateOption func(*truncateOptions)
+
+func WithTruncateHardDelete(hardDelete bool) func(*truncateOptions) {
+	return func(o *truncateOptions) {
+		o.HardDelete = hardDelete
+	}
+}
+
+func WithTruncateSkipPush(skipPush bool) func(*truncateOptions) {
+	return func(o *truncateOptions) {
+		o.SkipPush = skipPush
+	}
+}
+
+func WithTruncateMessage(message *Message) func(*truncateOptions) {
+	return func(o *truncateOptions) {
+		o.Message = message
+	}
+}
+
+func WithTruncateTruncatedAt(truncatedAt *time.Time) func(*truncateOptions) {
+	return func(o *truncateOptions) {
+		o.TruncatedAt = truncatedAt
+	}
+}
+
 // Truncate removes all messages from the channel.
-func (ch *Channel) Truncate() error {
+// You can pass in options such as hard_delete, skip_push
+// or a custom message.
+func (ch *Channel) Truncate(options ...TruncateOption) error {
+	// TODO: this thingy looks nice, but it's just a workaround to keep
+	// backward compatibility. Whenever we bump a major version, let's just
+	// modify this to: func (ch *Channel) Truncate(options *TruncateOptions)
```

Comment:
Not sure if I necessarily agree with this, but let's discuss it when we do :)

Reply:
I'm open to anything of course. :) I just like to keep things simple.

Reply:
I think it's fine as it is. Client is nicer

Reply:
that's why good to drop the comment

Reply:
And if you don't need any option you can just call it `Truncate()` instead of `Truncate(nil)`

Reply:
dropped comment 👍